packages/std/src/math/isqrt.rs

```diff
@@ -0,0 +1,53 @@
+use std::{cmp, ops};
+
+/// A trait for calculating the
+/// [integer square root](https://en.wikipedia.org/wiki/Integer_square_root).
+pub trait Isqrt {
+    /// The [integer square root](https://en.wikipedia.org/wiki/Integer_square_root).
+    fn isqrt(self) -> Self;
+}
+
+impl<
+        I: ops::Add<I, Output = I>
+            + ops::Div<I, Output = I>
+            + ops::Shr<u8, Output = I>
+            + cmp::PartialOrd
+            + Copy
+            + From<u8>,
```

Comment:
Nice, but can you move this to a [where](https://doc.rust-lang.org/rust-by-example/generics/where.html) clause?